src/v0/destinations/airship/utils.test.ts

```diff
@@ -0,0 +1,381 @@
+    originalTimestamp: '2025-01-07T10:57:38.768Z',
+    receivedAt: '2025-01-07T10:57:49.882Z',
+    request_ip: '77.248.183.43',
+    rudderId: 'b931e94b-3b22-462c-8b58-243cb4b37366',
+    sentAt: '2025-01-07T10:57:47.707Z',
+    userId: '1c5577e3-8d2d-4ecd-9361-88c2bfb254c5',
+  };
+  it('should return the correct attribute payload when jsonAttributes is not present in integrations object ', () => {
+    const message = {
+      ...commonEventProps,
+      context: {
+        ...commonContextProps,
+        traits: {
+          af_install_time: '2024-12-09 12:26:29.643',
+          af_status: 'Organic',
+          firstName: 'Orcun',
+          lastName: 'Test',
+          widgets_installed: ['no_widgets_installed', 'widgets_installed'],
+        },
+      },
+      event: 'identify',
+      integrations: {
+        All: true,
+      },
+      type: 'identify',
+    } as unknown as RudderMessage;
+
+    const expectedAttributePayload = {
+      attributes: [
+        {
+          action: 'set',
+          key: 'af_install_time',
+          value: '2024-12-09T12:26:29Z',
+          timestamp: '2025-01-07T10:57:38Z',
+        },
+        { action: 'set', key: 'af_status', value: 'Organic', timestamp: '2025-01-07T10:57:38Z' },
+        { action: 'set', key: 'first_name', value: 'Orcun', timestamp: '2025-01-07T10:57:38Z' },
+        { action: 'set', key: 'last_name', value: 'Test', timestamp: '2025-01-07T10:57:38Z' },
+        {
+          action: 'set',
+          key: 'widgets_installed[0]',
+          value: 'no_widgets_installed',
+          timestamp: '2025-01-07T10:57:38Z',
+        },
+        {
+          action: 'set',
+          key: 'widgets_installed[1]',
+          value: 'widgets_installed',
+          timestamp: '2025-01-07T10:57:38Z',
+        },
+      ],
+    };
+
+    const traits = getFieldValueFromMessage(message, 'traits');
+    const flattenedTraits = flattenJson(traits);
+    // @ts-expect-error error with type
+    const attributePayload = prepareAttributePayload(flattenedTraits, message);
+    expect(attributePayload).toEqual(expectedAttributePayload);
+  });
+
+  it('should throw error when jsonAttributes is present in integrations object and jsonAttribute(widgets_installed#123) is array', () => {
+    const message = {
+      ...commonEventProps,
+      context: {
+        ...commonContextProps,
+        traits: {
+          af_install_time: '2024-12-09 12:26:29.643',
+          af_status: 'Organic',
+          firstName: 'Orcun',
+          lastName: 'Test',
+          widgets_installed: ['no_widgets_installed', 'widgets_installed'],
+        },
+      },
+      event: 'identify',
+      integrations: {
+        All: true,
+        AIRSHIP: {
+          JSONAttributes: {
+            'widgets_installed#123': ['no_widgets_installed', 'widgets_installed'],
+          },
+        },
+      },
+      type: 'identify',
+    } as unknown as RudderMessage;
+
+    const expectedAttributePayload = {
```

Comment:
This is not being used as we are checking errors for this test case.

Reply:
This is done